plugins/modules/k8s_cluster_info.py

```diff
@@ -0,0 +1,185 @@
+  community.kubernetes.k8s_cluster_info:
+  register: api_status
+'''
+
+RETURN = r'''
+apis:
+  description:
+  - The API(s) that exists in dictionary
+  returned: success
+  type: dict
+  contains:
+    api_version:
+      description: API version
+      returned: success
+      type: str
+    categories:
+      description: API categories
+      returned: success
+      type: list
+    group_version:
+      description: Resource Group version
+      returned: success
+      type: str
+    kind:
+      description: Resource kind
+      returned: success
+      type: str
+    name:
+      description: Resource short name
+      returned: success
+      type: str
+    namespaced:
+      description: If resource is namespaced
+      returned: success
+      type: bool
+    preferred:
+      description: If resource version preferred
+      returned: success
+      type: bool
+    prefix:
+      description: Resource Path Prefix
+      returned: success
+      type: str
+    short_names:
+      description: Resource short names
+      returned: success
+      type: str
+    singular_name:
+      description: Resource singluar name
+      returned: success
+      type: str
+    urls:
+      description: Resource urls
+      returned: success
+      type: dict
+    available_api_version:
+      description: All available versions of the given API
+      returned: success
+      type: list
+    preferred_api_version:
+      description: Preferred version of the given API
+      returned: success
+      type: str
+'''
+
+
+import copy
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible_collections.community.kubernetes.plugins.module_utils.common import K8sAnsibleMixin, AUTH_ARG_SPEC
+try:
+    from kubernetes import client
+except ImportError:
+    # ImportError are managed by the common module already.
+    pass
+
+try:
+    try:
+        # >=0.10
+        from openshift.dynamic.resource import ResourceList
+    except ImportError:
+        # <0.10
+        from openshift.dynamic.client import ResourceList
+    HAS_K8S_INSTANCE_HELPER = True
+    k8s_import_exception = None
+except ImportError:
+    HAS_K8S_INSTANCE_HELPER = False
+    k8s_import_exception = traceback.format_exc()
+
+
+class KubernetesInfoModule(K8sAnsibleMixin):
+
+    def __init__(self):
+        module = AnsibleModule(
+            argument_spec=self.argspec,
+            supports_check_mode=True,
+        )
+        self.module = module
+        self.params = self.module.params
+
+        if not HAS_K8S_INSTANCE_HELPER:
+            self.module.fail_json(msg=missing_required_lib("openshift >= 0.6.2", reason="for merge_type"),
+                                  exception=k8s_import_exception)
+
+        super(KubernetesInfoModule, self).__init__()
+
+    def execute_module(self):
+        self.client = self.get_api_client()
+        results = {}
+        for resource in list(self.client.resources):
```

Comment:
We may want to expose the option to invalidate the cache (or invalidate it by default), as it's possible that APIs are added/removed without getting picked up (internally when you attempt to contact a specific API we will update properly, but when iterating over the APIs it may not pick up CRD changes to an existing API group). You can invalidate the cache with `self.client.resources.invalidate_cache()`

Reply:
Done.